tests/std/tests/GH_005780_non_ascii_locales/test.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <string>
+#include <clocale>
+#include <cassert>
+
+#include <xlocinfo>
+
+std::string set_locale(const std::string& locale_name) {
+    const char *ret = std::setlocale(LC_ALL, locale_name.c_str());
+    assert(ret != nullptr);
+    return ret;
+}
+
+std::string query_locale() {
+    const char *ret = std::setlocale(LC_ALL, nullptr);
+    assert(ret != nullptr);
+    return ret;
+}
+
+void test_gh_5780() {
+    // https://learn.microsoft.com/en-us/cpp/c-runtime-library/language-strings#supported-language-strings
+    std::string locale_name = set_locale("norwegian-bokmal.437");
+
+    const auto& char_not_ascii = [](char c){ return (c & 0x80) != 0; };

```

Comment:
There's no point to use lifetime-prolonging reference-binding-to-temporary here.

There's almost never need to use this obscure feature.

Just `const auto char_not_ascii` would be better.